reference/mongodb/architecture.xml

```diff
@@ -441,57 +441,57 @@ class AnotherClass1 implements MongoDB\BSON\Serializable {
   public $foo = 42;
   protected $prot = "wine";
   private $fpr = "cheese";
-  function bsonSerialize() {
+  function bsonSerialize(): array {
       return [ 'foo' => $this->foo, 'prot' => $this->prot ];
   }
 } // => { "foo" : 42, "prot" : "wine" }
 
 class AnotherClass2 implements MongoDB\BSON\Serializable {
   public $foo = 42;
-  function bsonSerialize() {
+  function bsonSerialize(): self {
```

Comment:
I realize the purpose of this example is to demonstrate an error, but is `self` even accepted here? I suppose since the interface uses `array|object`, this is an example of covariance?

Reply:
That is correct - `object` covers any object instance, which makes the return type valid.